src/bun.js/api/server.zig

```diff
@@ -1561,22 +1561,19 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
         }
 
         fn handleReject(ctx: *RequestContext, value: JSC.JSValue) void {
-            if (ctx.resp == null) {
+            if (ctx.isAbortedOrEnded()) {
                 ctx.finalizeForAbort();
                 return;
             }
+
             const resp = ctx.resp.?;
+            resp.clearAborted();
             const has_responded = resp.hasResponded();
             if (!has_responded)
                 ctx.runErrorHandler(
                     value,
                 );
 
-            if (ctx.isAborted()) {
```

Comment:
What if the request aborts after the error handler runs?

Reply:
fixed https://github.com/oven-sh/bun/pull/12472/commits/c7bcbe8d6835f06b122ae22f7081098d84064a6d